### PR TITLE
Mrc 5429 - warn on "..." specified for null driver in hipercow_init

### PR DESCRIPTION
**Pull request by weshinsley**

### CODE REVIEW

File: R/root.R [hunk 1]
```diff
@@ -24,6 +24,16 @@
 ##' path <- withr::local_tempfile()
 ##' hipercow_init(path)
 hipercow_init <- function(root = ".", driver = NULL, ...) {
+  if ((is.null(driver)) && (length(list(...)) > 0)) {
```
Comment (richfitz) [inline R/root.R:27]: Perhaps look into `rlang::check_dots_used` (https://rlang.r-lib.org/reference/check_dots_used.html) which might do this for us?

File: R/root.R [hunk 2]
```diff
@@ -24,6 +24,16 @@
...
+    args <- list(...)
+    cli::cli_alert_warning(
+      "driver was not specified, but extra args were detected: ")
+    ul <- cli::cli_ul()
```
Comment (richfitz) [inline R/root.R:31]: See comments in #134 about a better way to do these

**Review by richfitz — CHANGES_REQUESTED** [review]

File: R/root.R (hunk 2 above)
Comment (weshinsley) [reply R/root.R]: We decided to omit these...

File: R/root.R (hunk 1 above)
Comment (weshinsley) [reply R/root.R:27]: There is the possibility of... 

if (is.null(driver)) {
  rlang::check_dots_empty()
}

but converting this from an error to a warning, and including a message to say why 

! `...` must be empty.

only applies when driver = NULL was a bit fiddly.

**Review by richfitz — APPROVED** [review]

**Pull request merged**